pvanalytics/features/daytime.py

```diff
@@ -76,9 +76,9 @@ def _correct_edge_of_day_errors(night, minutes_per_value,
     # the day/night boundary at one end of the day - sunrise or sunset
     # - was correctly marked, it will be replaced with the rolling
     # median for that minute).
-    day_length = night.groupby(night.cumsum()).transform(
-        lambda x: len(x) * minutes_per_value
-    )
+    day_periods = (~night).astype(int)
+    day_length = 1 + day_periods.groupby(
+        night.cumsum()).transform('sum') * minutes_per_value
```

Comment:
```suggestion
    day_length = (1 + day_periods.groupby(
        night.cumsum()).transform('sum')) * minutes_per_value
```

Should it be this?

Reply:
Oops, you're right. I double checked the old outputs to the two variations above, and the updated version is the same:

![image](https://github.com/pvlib/pvanalytics/assets/70228568/c21d4666-afc8-490a-9c9d-99e8662a054b)